pycalphad/mapping/strategy/ternary_strategy.py

```diff
@@ -0,0 +1,396 @@
+    free_point = map_utils._generate_point_with_free_cs(point)
+
+    # Get normal and axis variable to step in (this will be along the maximum of the normal vector)
+    normal = _get_norm(free_point, axis_vars)
+    av = axis_vars[np.argmax(np.abs(normal))]
+    _log.info(f"Point {point.fixed_phases}, {point.free_phases} with normal {normal}. Testing derivative in {av}")
+
+    # This is here to have derivative with respect to the axis variable to follow the direction of the normal
+    #  NOTE: the derivative we compute is not the directional derivative with respect to the normal
+    #  but rather the derivative with respect to the axis variable, with the normal as a fixed condition
+    #  Since we lose the direction information of the normal when taking the derivative, mult_factor is here
+    #  to retain that information
+    mult_factor = np.sign(normal[axis_vars.index(av)])
+
+    # Create a linear combination that forces the stepping to be along the normal
+    lc, c = _create_linear_comb_conditions(point, axis_vars, normal)
+
+    # Create temporary set of conditions that replaces the fixed axis variable with the linear combination
+    del free_point.global_conditions[axis_vars[1-axis_vars.index(av)]]
+    free_point.global_conditions[lc] = c
+
+    # Test phase composition derivative for each phase stepping along the axis variable fixed along the normal
+    stable_cs = free_point.stable_composition_sets
+    cs_options = [[stable_cs[0], stable_cs[1]], [stable_cs[1], stable_cs[0]]]
+    options_tests = []
+    for cs_list in cs_options:
+        for av_test in axis_vars:
+            v_num = v.X(cs_list[1].phase_record.phase_name, av_test.species)
+            der = mult_factor*zeq.compute_derivative(free_point, v_num, av, free_den = False)
+            new_point = map_utils._generate_point_with_fixed_cs(point, cs_list[0], cs_list[1])
+            options_tests.append((der, new_point, av_test))
+
+    # Best index is the one with the largest derivative
+    best_index = -1
+    best_der = -1
+    for i in range(len(options_tests)):
+        _log.info(f"Option: Axis var {options_tests[i][2]}, derivative {options_tests[i][0]}, point {options_tests[i][1].fixed_phases}, {options_tests[i][1].free_phases}")
+        if abs(options_tests[i][0]) > best_der:
+            best_index = i
+            best_der = abs(options_tests[i][0])
+
+    if best_index == -1:
+        return options_tests[0][0], options_tests[0][1], options_tests[0][2], normal
+    else:
+        return options_tests[best_index][0], options_tests[best_index][1], options_tests[best_index][2], normal
+
+class TernaryStrategy(MapStrategy):
+    def __init__(self, dbf: Database, components: list[str], phases: list[str], conditions: dict[v.StateVariable, Union[float, tuple[float]]], **kwargs):
+        super().__init__(dbf, components, phases, conditions, **kwargs)
+        # TODO: I don't really like this since this assumes pure elements
```

Comment:
Resolve TODO

Reply:
I think it's okay to assume pure elements for now, and wait until the general Components are supported